server/message.go

```diff
@@ -0,0 +1,57 @@
+package server
+
+import (
+	"errors"
+
+	"github.com/coniks-sys/coniks-go/utils"
+)
+
+var (
+	ErrorMalformedClientMessage = errors.New("[coniks] Malformed client request")
+	ErrorNameExisted            = errors.New("[coniks] Registering identity is already registered")
+	ErrorInternalServer         = errors.New("[coniks] Internal server error")
+)
+
+const (
+	RegistrationType = "registration"
+)
+
+type Request struct {
+	Type    string
+	Request interface{}
+}
+
+type RegistrationRequest struct {
+	Username               string `json:"username"`
+	Key                    string `json:"key"`
+	KeyType                string `json:"key_type,omitempty"`
+	AllowUnsignedKeychange bool   `json:"allow_unsigned_key_change,omitempty"`
+	AllowPublicLookup      bool   `json:"allow_public_lookup,omitempty"`
+}
+
+type Response interface{}
+
+type ErrorResponse struct {
+	Error string
+}
+
+type RegistrationResponse struct {
+	TempraryBinding string `json:"tb"`
+}
+
+func NewErrorResponse(err error) *ErrorResponse {
+	return &ErrorResponse{Error: err.Error()}
+}
+
+func (reg *RegistrationRequest) serialize() []byte {
+	// serialize scheme:
```

Comment:
> I should rename ``serialization.go`` to something else. ``serialization.go`` includes functions that encode the messages between client & server

Yeah, I was a bit confused about the naming. Maybe something like ``messaging.go`` could work? I figured ``serialize()`` was used for something like the DB storage. We might want to move everything storage related to its own module?